### DISCUSSION TITLE
System Ordering Debugging?

### DISCUSSION THREAD

**Discussion by meowitzher** [bevyengine/bevy#5067] [Q&A] [answered]
Hello!

Is there a convenient way to print or see what the current system ordering is? Or simply some way to debug based on the ordering that Bevy chooses to run its systems in, I'm not picky! I've tried looking around, but all I've found is some fairly major ideas threads about reworking the way systems work, so I'm not certain what the current status is.

I currently have a problem where my project appears more jittery or unstable from one run to the next. Sometimes it runs very well, and I am attempting to figure out which ordering of the systems gives me that performance so that I can know which systems I need to enforce an ordering on using `before` and `after`. Other times the moving sprites tends to jitter and not be smooth, which I'd like to eliminate.

I've ordered many of my systems based on what I can deduce about how the code should work, but I can't see how any of my remaining unordered systems should affect jittering, so I'd like to be able to look at the system ordering during the "good" runs to help me find the mistake I am making.

**Comment by hymm** [comment, marked as answer]
You can use https://docs.rs/bevy/latest/bevy/ecs/schedule/struct.ReportExecutionOrderAmbiguities.html to detect systems that should potentially be ordered.

Also you can use tracing to see the actual execution order. https://github.com/bevyengine/bevy/blob/main/docs/profiling.md

**Reply by meowitzher** [reply]
Thanks for both of these excellent resources!

It turns out I was doing some stuff in the PostUpdate stage without clarifying ambiguity with Bevy's native transform propagation system. Just making sure that ran afterwards fixed the issue.
